File: Examples/CPP/CPP_LinkedList_Include_FR/List.cpp
#include <iostream>
using namespace std;
#include "List.h"
#include "Node.h" //  we cannot use FR

//#include "IntItem.h"
//class IntItem;





List::List() {
    head = nullptr;
}

void List::print() {
    Node* currNode = head;
    cout << "The list contains: ";
    while (currNode != nullptr ) {
        currNode->print();
        cout << " ";
        currNode = currNode->getNext();
    }
    cout << endl;
}

void List::insert(IntItem* item) {
    Node* newNode = new Node(item);
    newNode->setNext(head);
    head = newNode;
}

File: Examples/Java/List.java
public class List extends ListItem {
    private Node head;
    public List() {
        head = null;
    }

    public void print() {
        System.out.print("[");
        if (head != null) {
            head.print();
        }
        System.out.print("]");
    }

    public void add ( ListItem data ) {
        head = new Node(data, head);
    }
}

